Common/Product/TestAdapter/VisualStudioApp.cs

```diff
@@ -75,7 +77,34 @@ class VisualStudioApp : IDisposable {
             return dte;
         }
 
+#if DEV15
+        private static bool DTELoaded = false;
+#endif
+
         private static DTE GetDTE(int processId) {
+#if DEV15
+            // VS 2017 doesn't install some assemblies to the GAC that are needed to work with the
+            // debugger, and as the tests don't execute in the devenv.exe process, those assemblies
+            // fail to load - so load them manually from PublicAssemblies.
+
+            // Use the executable name, as this is only needed for the out of proc test execution
+            // that may interact with the debugger (vstest.executionengine.x86.exe).
+            if (!DTELoaded)
+            {
+                string currentProc = Process.GetCurrentProcess().MainModule.FileName;
+                if (Path.GetFileName(currentProc).ToLowerInvariant().Equals("vstest.executionengine.x86.exe"))
```

Comment:
prefer `StringComparer.OrdinalIgnoreCase.Equals` since this deals with nulls cleanly. (i.e. doesn't throw if either argument is null)